docs/porting-a-new-endpoint.md

```diff
@@ -0,0 +1,233 @@
+| MlflowService          | deleteTag                    |     No      |
++------------------------+------------------------------+-------------+
+```
+
+## Enable the Endpoint
+
+Add the missing endpoint to [endpoints.go](../magefiles/generate/endpoints.go) under the correct service. After that, run `mage generate`.
+
+⚠️ This should have updated a few files that you are not expected to update yourself! ⚠️
+
+### New fiber route
+
+In [pkg/server/routes/tracking.g.go](../pkg/server/routes/tracking.g.go), a new route was added:
+
+```go
+	app.Post("/mlflow/runs/delete-tag", func(ctx *fiber.Ctx) error {
+		input := &protos.DeleteTag{}
+		if err := parser.ParseBody(ctx, input); err != nil {
+			return err
+		}
+		output, err := service.DeleteTag(utils.NewContextWithLoggerFromFiberContext(ctx), input)
+		if err != nil {
+			return err
+		}
+		return ctx.JSON(output)
+	})
+```
+
+This means that our Go server will now process the API request.
+
+### New Service Method
+
+In [pkg/contract/service/tracking.g.go](../pkg/contract/service/tracking.g.go), the interface has a new function that we need to implement, which is used in the routes file above.
+
+```go
+type TrackingService interface {
+    ...
+	DeleteTag(ctx context.Context, input *protos.DeleteTag) (*protos.DeleteTag_Response, *contract.Error)
+}
+```
+
+### New FFI Endpoint
+
+In [pkg/lib/tracking.g.go](../pkg/lib/tracking.g.go), a new FFI endpoint was generated.
+
+```go
+//export TrackingServiceDeleteTag
+func TrackingServiceDeleteTag(serviceID int64, requestData unsafe.Pointer, requestSize C.int, responseSize *C.int) unsafe.Pointer {
+	service, err := trackingServices.Get(serviceID)
+	if err != nil {
+		return makePointerFromError(err, responseSize)
+	}
+	return invokeServiceMethod(service.DeleteTag, new(protos.DeleteTag), requestData, requestSize, responseSize)
+}
+```
+
+We will later create a Python binding to enable us to invoke this generated Go code.
+
+## Validate Input
+
+A first step to port an endpoint would be to check the request validation happening at the HTTP level. Open [handlers.py](../.mlflow.repo/mlflow/server/handlers.py) (from the `.mlflow.repo`) to see which fields are required.
+
+```python
+@catch_mlflow_exception
+@_disable_if_artifacts_only
+def _delete_tag():
+    request_message = _get_request_message(
+        DeleteTag(),
+        schema={
+            "run_id": [_assert_required, _assert_string],
+            "key": [_assert_required, _assert_string],
+        },
+    )
+    _get_tracking_store().delete_tag(request_message.run_id, request_message.key)
+    response_message = DeleteTag.Response()
+    response = Response(mimetype="application/json")
+    response.set_data(message_to_json(response_message))
+    return response
+```
+
+We notice that `run_id` and `key` are required. Too avoid writing repeatitive code in Go, we make use of [Go validator](https://github.com/go-playground/validator) and use annotations to validate the incoming structs.
```

Comment:
```suggestion
We notice that `run_id` and `key` are required. To avoid writing repetitive code in Go, we make use of [Go validator](https://github.com/go-playground/validator) and use annotations to validate the incoming structs.
```